samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -73,34 +78,33 @@ public ZkJobCoordinator(Config config, MetricsRegistry metricsRegistry) {
     this.config = config;
     ZkConfig zkConfig = new ZkConfig(config);
     ZkKeyBuilder keyBuilder = new ZkKeyBuilder(new ApplicationConfig(config).getGlobalAppId());
+    ZkClient zkClient = ZkCoordinationServiceFactory
+        .createZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
```

Comment:
done. 
Also changed GenIZkChildListener from ZkUtils (from the previous comment, but github doesn't let me comment)